packages/gatsby/src/utils/api-node-docs.js

```diff
@@ -181,6 +181,11 @@ exports.modifyBabelrc = true
  */
 exports.modifyWebpackConfig = true
 
+/**
+ * Called before the config is validated and the cache is initialized.
```

Comment:
Maybe it's worth starting with conveying what You wrote in PR description: 

> This PR adds a new hook which gets triggered basically before Gatsby does anything.

and then note that this hook is different than `onPreBootstrap` because it's

> Called before the config is validated and the cache is initialized 

?

Reply:
Alright, I'll align the description :)